src/main/java/org/variantsync/diffdetective/analysis/strategies/AnalyzeAllAndExport.java

```diff
@@ -0,0 +1,48 @@
+package org.variantsync.diffdetective.analysis.strategies;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+
+import org.apache.commons.io.output.CloseShieldOutputStream;
+import org.tinylog.Logger;
+import org.variantsync.diffdetective.datasets.Repository;
+import org.variantsync.diffdetective.diff.CommitDiff;
+import org.variantsync.diffdetective.util.IO;
+
+/**
+ * Collects all linegraph representations generated by an analysis and exports them at the end.
+ * @author Benjamin Moosherr
+ */
```

Comment:
From this documentation, the differrence to `AnalyzeAllThenExport` is not clear to me. It seems as if both are the same?

Reply:
Where is this new class needed?

Reply:
They differ in their flushing behavior: `AnalyzeAllThenExport` keeps all line graphs in memory until the analysis for all commits has finished and then writes it to disk. In contrast `AnalyzeAllAndExport` writes everything directly to disk (and flushes after every commit).

This is also an easy way to save memory, so I changed the `AnalysisStrategy` in `DiffTreeMiner` from `AnalyzeAllThenExport` to `AnalyzeAllAndExport`. I actually forgot to ask that last week: Is this change even desired? I ask, because `AnalyzeAllThenExport` seemed to be written deliberately to export the line graphs only if the analysis finished successfully.

This comment is actually wrong :sweat_smile:. Will change that soon.

Reply:
The goal was to reduce IO operations in favor of a potentially bigger main memory footprint. Otherwise we have an IO operation after each of millions of commits. AnalyzeAllThenExport thus exported commit linegraphs in batches of 1000 commits.

Reply:
Both strategies are useful. Please keep the old one in DiffTreeMiner for now and refactor your comment here into documentation of the strategies.

Reply:
The `DiffTreeMiner` is now using `AnalyzeAllThenExport` again and the comment of `AnalyzeAllAndExport` is referencing `AnalyzeAndExportIncrementally` for clarity.